src/Transport/Config/ExtensionPoints/AzureServiceBusTopologySettings.cs

```diff
@@ -1,14 +1,48 @@
 ﻿namespace NServiceBus
 {
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+    using NServiceBus.AzureServiceBus;
+    using NServiceBus.AzureServiceBus.TypesScanner;
     using NServiceBus.Settings;
 
     public class AzureServiceBusTopologySettings : TransportExtensions<AzureServiceBusTransport>
     {
+        private readonly SettingsHolder _settings;
+
         public AzureServiceBusTopologySettings(SettingsHolder settings)
             : base(settings)
         {
+            _settings = settings;
+        }
+
+        public AzureServiceBusTopologySettings RegisterPublisherForType(string publisherName, Type type)
+        {
+            AddScannerForPublisher(publisherName, new SingleTypeScanner(type));
+            return this;
         }
 
+        public AzureServiceBusTopologySettings RegisterPublisherForAssembly(string publisherName, Assembly assembly)
+        {
+            AddScannerForPublisher(publisherName, new AssemblyTypesScanner(assembly));
+            return this;
+        }
 
+        private void AddScannerForPublisher(string publisherName, ITypesScanner scanner)
```

Comment:
We designed our config API to be fluent. Isn't better to give to users only allowed methods for selected topology?
There is an extension point to allow users to add a custom topology implementation. I thought that adding a property to `ITopology` interface we  can decide if a topology needs to configure `publisher names` vs. `event type` mapping or not.